### PR TITLE
Replaces sleeping carp pierce immunity with wound resistance, few other small fixes

### LINKED ISSUE #51218: Health analyzer doesn't display health scan while blind
Issue reported from  Round ID: [137718](https://scrubby.melonmesa.com/round/137718) (/tg/Station Basil [ENGLISH] [US-WEST] [100% FREE LAG])

Reporting client version: 513.1512

While blind, attempting to use a health scanner in "check physical health" mode on someone while blind, nothing is outputted. Output is shown as usual for "scan chemical contents" or "report extra info on wounds." This should either show as usual for "check physical health" mode or show something in the chat explaining that I can't use the health analyzer while blind.

### LINKED ISSUE #53191: Changling Reviving Stasis bug
Issue reported from  Round ID: [144969](https://scrubby.melonmesa.com/round/144969) (/tg/Station Manuel [ENGLISH] [US-WEST] [ROLEPLAY] [99% LAG FREE])

Reporting client version: 513.1528

I was using reviving stasis to heal some wounds, I put myself in a locker to hide then used the ability. While in stasis I reset my genomes and it disabled my ability to actually revive. near as I can tell it "reset" the stasis ability preventing the revive option from triggering. I tried having the admin kill my guy then revive but that didnt break it either.

### LINKED ISSUE #53733: you can't inject someone's arm if they're wearing a thickmaterial vest
```
/mob/living/carbon/human/can_inject(mob/user, error_msg, target_zone, penetrate_thick = FALSE, ignore_species = FALSE)
	. = TRUE // Default to returning true.
	if(user && !target_zone)
		target_zone = user.zone_selected
	// we may choose to ignore species trait pierce immunity in case we still want to check skellies for thick clothing without insta failing them (wounds)
	if(ignore_species)
		if(HAS_TRAIT_NOT_FROM(src, TRAIT_PIERCEIMMUNE, SPECIES_TRAIT))
			. = FALSE
	else if(HAS_TRAIT(src, TRAIT_PIERCEIMMUNE))
		. = FALSE
	// If targeting the head, see if the head item is thin enough.
	// If targeting anything else, see if the wear suit is thin enough.
	if (!penetrate_thick)
		if(above_neck(target_zone))
			if(head && istype(head, /obj/item/clothing))
				var/obj/item/clothing/CH = head
				if (CH.clothing_flags & THICKMATERIAL)
					. = FALSE
		else
			if(wear_suit && istype(wear_suit, /obj/item/clothing))
				var/obj/item/clothing/CS = wear_suit
				if (CS.clothing_flags & THICKMATERIAL)
					. = FALSE
	if(!. && error_msg && user)
		// Might need re-wording.
		to_chat(user, "<span class='alert'>There is no exposed flesh or thin material [above_neck(target_zone) ? "on [p_their()] head" : "on [p_their()] body"].</span>")
```
can_inject() doesn't check to see if the item you're wearing in your suit slot actually covers the non-head bodypart you're targetting. Thus, you can't inject someone who's wearing a syringe-proof vest or whatever that explicitly doesn't cover their arms in their arm.

### LINKED ISSUE #53843: Sleeping carp users can't heal their wounds using conventional methods
For some reason, https://github.com/tgstation/tgstation/pull/53307 only accounts for species with the TRAIT_PIERCEIMMUNE trait, not other sources of the TRAIT_PIERCEIMMUNE trait, like sleeping carp.

This means that sleeping carp users have very few options (other than cryo) for dealing with, say, broken legs.

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
Back in #50904, wesoda read me like a textbook

> So basically this is the trigger to a pr balance war. Its too radical but will get people scared enough to accept a PR which gives us a weaker form of original carp (lemme guess you can’t aggressive grab people?). Nice job

I wasn't really thinking about the specifics of how sleeping carp should function, I just made that PR to apply pressure to get something I thought was obnoxious changed. After it was merged, I realized that leaving in the pierce immunity was actually an unintentional mistake on my part since they still couldn't use lots of medical stuff despite losing the resists, but I figured someone else would realize I botched it and fix it. Turns out no one ever did!

So this is me redeeming for my sins and finally removing the pierce immunity from sleeping carp, so they can now use medical supplies as normal. In its place, I gave sleeping carp the wound resistance trait, which reduces the wounding power of all incoming attacks by a decent amount. I figure it makes sense cause it gives the user more staying power and reduces how easy it is to knock them out of commission with a lucky hit. Please don't start doing this everywhere for all of your pet antags.

* Fixes: #53843

I also fix a few other things, like-
* Fixes: #53733
Looking back I'm pretty sure I got memed by this issue because afaict the only clothing that is THICKMATERIAL and doesn't cover the entire body below the neck is the bee costume which leaves the legs uncovered, which means this was a waste, but can_inject() now checks the specific targeted bodypart to see if it's covered rather than just checking their suit.

* Fixes: #53191
You can no longer readapt while you're in stasis, preventing this bug

* Fixes: #51218
Medical, chemical, and wound scanners are now properly inaccessible to the blind. Progress!

## Changelog
:cl: Ryll/Shaps
balance: Sleeping carp users are no longer immune to piercing, which means they can once again use most medical supplies, but will trip on glass and such like normal people. In return, they are now somewhat more resistant to suffering wounds.
tweak: Medical, chemical, and wound scanners are now consistent in that they are all inaccessible to the blind. Progress!
tweak: Thick material checks (like syringes and such) are now applied on a bodypart by bodypart basis, rather than just depending on your suit.
fix: As a changeling, you can no longer readapt your mutations while in fakedeath stasis, which would cause you to stay in sleepies forever
/:cl:

### CODE REVIEW

**Comment by Cyprex** [comment]
Wound immunity I would value at maybe 8 TC at best, and only under very specific circumstances.

**Comment by necromanceranne** [comment]
Thank you for fixing this.

**Comment by Ryll-Ryll** [comment]
> Wound immunity I would value at maybe 8 TC at best, and only under very specific circumstances.

You still get the whole martial art thing bro

Also mind that it's not wound immunity, it's a reduction in incoming wounding power, which means... well, I'm not really sure! Edge added the trait in as a reverse of the easier wounding trait I added for kicks, but I didn't really plan for modifiers like this when I wrote the system, so I actually have no clue how much this affects taking wounds, and whether it makes you practically immune, or just more resilient. I'll do a few tests to get a rough figure I guess.

File: code/game/objects/items/devices/scanners.dm [hunk 1]
```diff
@@ -412,6 +416,10 @@ GENE SCANNER
 	to_chat(user, jointext(render_list, ""), trailing_newline = FALSE) // we handled the last <br> so we don't need handholding
 
 /proc/chemscan(mob/living/user, mob/living/M)
+	if(user.is_blind())
+		to_chat(user, "<span class='warning'>You realize that your scanner has no accessibility support for the blind!</span>")
+		return
```
Comment (Rohesie) [inline code/game/objects/items/devices/scanners.dm:421]: This is a good change, but is it related?
Could you atomize these fixes in their own PRs?

File: code/modules/antagonists/changeling/changeling.dm [hunk 2]
```diff
@@ -207,6 +207,9 @@
 	if(!ishuman(owner.current))
 		to_chat(owner.current, "<span class='warning'>We can't remove our evolutions in this form!</span>")
 		return
+	if(HAS_TRAIT_FROM(owner.current, TRAIT_DEATHCOMA, "changeling"))
+		to_chat(owner.current, "<span class='warning'>We are too busy reforming ourselves to readapt right now!</span>")
+		return
```
Comment (Rohesie) [inline code/modules/antagonists/changeling/changeling.dm:212]: Same for this, it looks like a standalone fix?

**Comment by Ryll-Ryll** [comment]
Yeah I'll do it tonight

File: code/game/objects/items/devices/scanners.dm [hunk 3]
```diff
@@ -459,6 +467,10 @@ GENE SCANNER
 	if(!istype(patient))
 		return
 
+	if(scanner && user.is_blind())
```
Comment (ShizCalev) [inline code/game/objects/items/devices/scanners.dm:470]: atomize as rohesie said. also use `user.can_read()` instead.

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by Iamgoofball** [comment]
?

**Pull request closed without merging**